### ISSUE TITLE
Final or best result

### ISSUE THREAD

**Issue by JingTongsh** [SLDGroup/EMCAD#19]
Hello, I have run your training code without modifying any hyper-parameters on the Synapse dataset.
I have only run it once with `encoder=pvt_v2_b2, seed=2022`, and got `final dice=0.8092, best dice=0.8396`.
The dice score reported in your paper for `encoder=pvt_v2_v2` is `dice=0.8363`, "averaged over five runs".
Considering the performance gap, my questions are:
1. Did you report the best or the final results in your paper? Or should I run more times?
2. Could you provide your results from 5 runs and corresponding random seeds?

**Comment by mostafij-rahman** [comment]
Hi,

Thank you for your comment. We reported the best DICE scores averaged over five runs. Due to randomness in weight initialization, you may get a better or lower DICE score in different runs. It is evident from your run that EMCAD produces 83.96% DICE score which is even better than our reported DICE score.

You can report results from our paper if you use the same parameters.

Best,
Mostafij

**Comment by JingTongsh** [comment]
Thank you for answering, but I don't think it's reasonable to report the best results.
- It's quite possible that the DICE score is just bouncing up and down before converging, and somehow reached a nice peak.
- The best result is not applicable. You have full access to the validation set throughout the training procedure. But in practice, the test set is hardly labeled. 
- The best result may not generalize well. Have you tried the official Synapse test set, instead of the version provied by TransUNet?

**Comment by JingTongsh** [comment]
By the way, why don't you compare with nnUNet? It gets 84.03% DICE score on your version of Synapse dataset.

**Comment by mostafij-rahman** [comment]
Thanks for your comments. 

First, we would like to let you know that we report the average DICE score of five runs. We are not the first to report the best DICE score; it's a very common and well-established practice and many other methods including TransUNet reported the best DICE score on the validation set. We followed the experimental protocols of established methods for fair comparisons.

Second, we didn't compare our method with nnUNet because nnUNet uses a different preprocessing, training, and evaluation pipeline and our EMCAD is not directly comparable to nnUNet. If our method is integrated into the nnUNet framework in place of UNet architecture, we believe our method will produce better DICE scores than nnUNet with significantly lower computational costs.

**Comment by wangru1026** [comment]
Hi Mostafij, could you please provide the testing code for this part? It would really help in verifying and reproducing the results. Thanks a lot!

**Comment by wangru1026** [comment]
Hi Mostafij, Thank you for sharing the test code. I was able to reproduce the results successfully. The model performs very well. I will continue to study it in depth and look forward to more of your work!
